app/models/manageiq/providers/regions.rb

```diff
@@ -0,0 +1,46 @@
+module ManageIQ::Providers
+  class Regions
+    class << self
+      def regions
+        from_file.merge(additional_regions).except(*disabled_regions)
+      end
+
+      def all
+        regions.values
+      end
+
+      def names
+        regions.keys
+      end
+
+      def find_by_name(name)
+        regions[name]
+      end
+
+      private
+
+      def from_file
+        # Memoize the regions file as this should not change at runtime
+        @from_file ||= YAML.load_file(regions_yml).transform_values(&:freeze).freeze
+      end
+
+      def additional_regions
+        Settings.dig(:ems, ems_type, :additional_regions).to_h.stringify_keys
+      end
+
+      def disabled_regions
+        Settings.dig(:ems, ems_type, :disabled_regions).to_a
+      end
+
+      def ems_type
+        vendor = module_parent.name.sub("ManageIQ::Providers::", "").sub("::", "_").downcase
+
+        "ems_#{vendor}".to_sym
```

Comment:
I thought ems_type was something we could get from the EMS.ems_type method

Reply:
This is a level above the managers so we don't have access to e.g. `...Amazon::CloudManager.ems_type`

Reply:
Interesting... I was going to suggest maybe we should expose it at that higher level, but then again a single provider could have multiple ems_types.

Reply:
Yeah, I think this gets back to what is the ems vs what is the provider

IMO

AWS - provider
   - EC2
   - EBS
   - S3
   - Network

So ems_type is per-manager but we need something a level up.  That is what this would track since regions are shared across all services.

Not sure we're going to solve that in this PR :laughing: